src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingStructureBuilder.java

```diff
@@ -62,6 +60,11 @@
      */
     private Map<String, BuildingBuilderResource> neededResources = new LinkedHashMap<>();
 
+    /**
+     * The different possible buckets.
+     */
+    private Deque<Tuple<Map<String, Integer>, Integer>> buckets = new ArrayDeque<>();
```

Comment:
Instead of doing `Deque<Tuple<Map<String, Integer>, Integer>>` you should make this an actuall class with documentation what what is. So create something like `StructureBuildSection` and make this field a `Deque<StructureBuildSection>` Patch all your fields and methods accordingly, and do not forget to document.